src/components/Avatar/Avatar.tsx

```diff
@@ -53,7 +53,28 @@ export interface Props {
   variant?: 'icon' | 'initials' | 'image';
 }
 
-function getInitials(fromName: string): string {
+/**
+ * Algo for names with special characters in:
+ * - Split name parts by space
+ * - for a given part,
+ *  - if that part has emoji or extended characters, {Action?}
+ *  - if that part does not, then use the first ASCII character
+ * @param str
+ * @returns
+ */
+function produceAbbreviation(str: string): string {
+  // unicode character range for surrogate pairs (first and second halves)
+  if (str.match(/([\uD800-\uDFFF])*/g)?.filter((match) => !!match).length) {
+    // grab first part if string has a surrogate chunk after the first character
+    // this assumes that the first part of the surrogate pair is legible
+    // TODO: See sample cases where matching splits surrogates improperly
+    return str.slice(0, 2);
```

Comment:
After some internet research, splitting emoji seems hard.

We could try https://github.com/flmnt/graphemer, see if it passes the compound test cases.

Reply:
[This sequence alone](https://github.com/flmnt/graphemer/blob/master/src/Graphemer.ts#L113-L122) tells me i probably should stop trying to solve this myself lol